File: 1erdepartamental/python/10.promedio_puntero/main.py
"""
Programa: Promedio con Puntero
Demuestra simulación de punteros con iterador encapsulado
"""
#Clase Principal
from core.estadisticas import EstadisticasPuntero
#Funcion Principal
def main ():
    e = EstadisticasPuntero([95, 25, 5, 100, 5])

    print(f"Suma: {e.suma()}")
    print(f"Promedio: {e.promedio()}")
    print(f"Máximo: {e.maximo()}")
    print(f"Mínimo: {e.minimo()}")

#Punto de Entrada
if __name__ == "__main__":
    main()

File: 1erdepartamental/c++/10.promedio_puntero/main.cpp
#include <iostream>
#include "core/EstadisticasPuntero.h"

int main() {
    // Crea estadisticas usando punteros
    EstadisticasPuntero e({10, 25, 5, 30, 15});
    
    // Muestra resultados
    std::cout << "Suma: " << e.suma() << std::endl;
    std::cout << "Promedio: " << e.promedio() << std::endl;
    std::cout << "Máximo: " << e.maximo() << std::endl;
    std::cout << "Mínimo: " << e.minimo() << std::endl;
    
    return 0;
}
